src/backend/distributed/planner/recursive_planning.c

```diff
@@ -111,17 +115,60 @@ typedef struct VarLevelsUpWalkerContext
 } VarLevelsUpWalkerContext;
 
 
+/*
+ * NonColocatedSubqueryContext is used to recursively plan subquery
+ * joins that are not on the distribution column.
+ */
+typedef struct NonColocatedSubqueryContext
```

Comment:
Since it doesn't just concern subqueries, should we call it `NonColocatedJoinContext`? (also wondering if there is a different word than context that applies here).